Sprint 16 task

s4fonovoleg opened the pull request.

s4fonovoleg committed:
Sprint 16 task

amardanov commented on Tracker/Category/CategoriesViewModel.swift:
```diff
@@ -0,0 +1,56 @@
+	// MARK: Lifecycle
+	
+	init() {
+		initCategories()
```
Не забывай про форматирование кода в проекте. Отступы можно поправить с помощью сочетаний клавиш `cmd+a`, `control+i`. Советую пройтись и это сделать в каждом файле

s4fonovoleg replied:
Конкретно в этом фале комбинация cmd+a, control+i не меняет ничего (специально проверил ее работоспособность добавив лишние табы), но буду иметь ввиду

P.S. возможно гитлаб показывает таб длиннее, чем четыре пробела? и потому кажется что форматирование съехало

amardanov commented on Tracker/CreateTracker/CreateTrackerTableViewCell.swift:
```diff
@@ -0,0 +1,8 @@
+//
```
нужен ли этот файл?

s4fonovoleg replied:
случайно затесался (отменял изменения в гите - он оставил файл)

amardanov commented on Tracker/CreateTracker/CreateTrackerViewController.swift:
```diff
@@ -360,15 +423,60 @@ final class CreateTrackerViewController: UIViewController {
 	// MARK: Private methods
 	
+	private func getWeekDaysShortNames() -> String {
+		var shortNames = weekDays.map { item in
```
можно записать такое в одно выражение 
```
weekDays
            .map { weekDayShortName[$0] ?? "" }
            .joined(separator: ", ")
```

s4fonovoleg replied:
оставлял так в угоду читаемости, хотя она здесь, наверное, не пострадает

amardanov commented on Tracker/OnboardingViewController.swift:
```diff
@@ -0,0 +1,173 @@
+// MARK: UIPageViewControllerDataSource
+
+extension OnboardingViewController: UIPageViewControllerDataSource {
+	func pageViewController(_ pageViewController: UIPageViewController, viewControllerBefore viewController: UIViewController) -> UIViewController? {
```
Рекомендую придерживаться следующего стиля написания длинных вызовов и длинных объявлений функций. Как пример можешь использовать то, что я напишу. Такой формат принят за основу во многих компаниях и следование ему даст несколько очков к карме при работе над боевыми проектами

```
func pageViewController(
        _ pageViewController: UIPageViewController,
        viewControllerBefore viewController: UIViewController
    ) -> UIViewController? {
```

s4fonovoleg replied:
поправил, спасибо за пример

s4fonovoleg committed:
Удален лишний пустой файл
Исправлено форматирование
Добавлен пропуск экрана онбординга при повторном запуске приложения
